web/src/peer_data.ts

```diff
@@ -29,12 +30,31 @@ const pending_subscriber_requests = new Map<
 export function clear_for_testing(): void {
     stream_subscribers.clear();
     fetched_stream_ids.clear();
+    pending_subscriber_requests.clear();
 }
 
 const fetch_stream_subscribers_response_schema = z.object({
     subscribers: z.array(z.number()),
 });
 
+// This function will always resolve to a LazySet but could hang
+// indefinitely trying to fetch data. `num_attempts` should only
+// be set in the recursive call.
```

Comment:
Are we okay with this hanging indefinitely? Or would we rather have a `max_num_attempts` and eventually return `null` if we can't get the subscribers? I feel like having a max number of attempts is reasonable, especially since the user will likely eventually navigate away from the relevant view and won't need the data anymore (for the time being) anyways.

Reply:
I think the scenario to think about is where the user is using Zulip offline. It's probably fine for this to retry indefinitely in that scenario in some sense, but it might not be very friendly if it comes back online and then hammers the server with 1 query per channel the user visited while offline.

Reply:
I guess because of the backoff for retries, the queries will likely be a bit spread out? Though not that necessary to fetch anymore, of course.

Reply:
I'm inclined to keep it as it is for now, since I don't feel a strong preference and that feels like the simpler solution, but if anyone else has a preference for doing something differently I'm happy to work on that.